packages/block-editor/src/components/observe-typing/index.js

```diff
@@ -156,67 +135,45 @@ class ObserveTyping extends Component {
 			return;
 		}
 
-		onStartTyping();
+		startTyping();
 	}
 
 	/**
 	 * Stops typing when focus transitions to a non-text field element.
 	 *
 	 * @param {FocusEvent} event Focus event.
 	 */
-	stopTypingOnNonTextField( event ) {
+	function stopTypingOnNonTextField( event ) {
 		event.persist();
```

Comment:
I know it was here before, but this could be avoided if you extract the target here.

Reply:
> I know it was here before, but this could be avoided if you extract the target here.

It makes me anxious if it was done this way on purpose originally, since it seems an obvious solution 😅 Could have been an oversight.

(IIRC, I might have been the one who wrote this code originally)

Reply:
It was introduced originally in #5552. The "Testing instructions" there are still valid even without persisting the event, so I think it should be a safe change. Updated in 1ce296973cbdb5228bafeb6d947540a012631b36.

Reply:
I'm a bit curious whether the previous implementation could have had a negative performance impact. Presumably if React felt need to reuse from a pool of events, there might be some "leakiness" by forcing each of these to persist forever. I'd like to think they could be garbage collected if there's nothing ever referencing it, but that depends if the runtime considers the timeout callback closure as a perpetual reference.

Reply:
Yeah, I think we should try to avoid it. It seems to be possible most of the time.